### PR TITLE
AMD Builds / Travis-CI

### PR DESCRIPTION
**Pull request by GordonSmith**
Add AMD Builds.
Add Travis-CI support.

Signed-off-by: Gordon Smith gordon.smith@lexisnexis.com

### CODE REVIEW

**Comment by GordonSmith** [comment]
@buunguyen - Here is my AMD build + Travis support - the demos work is still in progress, but if you can eyeball my gulpfile / travis changes to see if they are ok?

**Comment by GordonSmith** [comment]
@buunguyen Updated with working demos.

File: bower.json [hunk 1]
```diff
@@ -27,18 +27,15 @@
     "rjs.noop.js"
   ],
   "dependencies": {
-    "requirejs": "2.1.17",
-    "requirejs-plugins": "1.0.3",
-    "require-css": "0.1.8",
-    "c3": "0.4.9",
-    "colorbrewer": "1.0.0",
-    "d3": "3.5.0",
-    "dagre": "0.7.1",
```
Comment (GordonSmith) [inline bower.json:36]: You may want this version of dagre?  As I mentioned before, the bower dagre is a bit broken, so I have included the "pre-built" version in "third_party" which makes my life a lot easier - we can include it here in bower as well?

Comment (buunguyen) [inline bower.json:36]: No, I don't want any particular version. It's fine to include another version if it's in a git repo.

**Comment by buunguyen** [comment]
LGTM, just some minor suggestions, no big deal.

**Comment by GordonSmith** [comment]
I am now grabbing dagre from github directly which cleaned up  a bunch of gulp code.  Ony weirdness is that bower downloads it as index.js (there is an open bug report about it...) but I just used paths to fix.

**Comment by buunguyen** [comment]
@GordonSmith when I ran `gulp`, I got this error:

```
[14:01:34] 'amd_layers' errored after 46 ms
[14:01:34] Error: Error: ENOENT, no such file or directory '/Projects/hms/hms/Visualization/dist/amd/lib/d3/d3.js'
In module tree:
    viz/marshaller/Graph
```

I'm not sure why it expects d3 in `/dist/amd/lib`. Is there a Gulp task to copy files from `bower_components` to `dist/amd/lib`?

**Comment by buunguyen** [comment]
@GordonSmith never mind, I figured it out, I didn't install bower_components in root. Build runs good now!

**Pull request merged**